doc/manual_experimental.rst

```diff
@@ -1157,6 +1157,22 @@ the ordinary AST predicates:
     a = b
     inc a, c
 
+Another example:
+
+.. code-block:: nim
+  proc somefunc(s: string)                 = assert s == "variable"
+  proc somefunc(s: string{nkStrLit})       = assert s == "literal"
+  proc somefunc(s: string{nkRStrLit})      = assert s == r"raw"
+  proc somefunc(s: string{nkTripleStrLit}) = assert s == """triple"""
+  proc somefunc(s: static[string])         = assert s == "constant"
+  var variable = "variable"
+  somefunc(variable)
+  const constant = "constant"
+  somefunc(constant)
+  somefunc("literal")
+  somefunc(r"raw")
+  somefunc("""triple""")
```

Comment:
I think this would benefit from a small comment in the code block that explains what the example does.  
Something like:
```
# Use parameter constraints to provide overloads based on both the input parameter type and form.
```